src/vs/workbench/api/common/extHost.protocol.ts

```diff
@@ -1978,6 +1979,7 @@ export interface ICodeActionDto {
 	isPreferred?: boolean;
 	isAI?: boolean;
 	disabled?: string;
+	editRanges?: Range[];
```

Comment:
should this be IRange[]?

Reply:
Yes in the protocol you always want to use the serializable types (such as `IRange`) instead of the classes that implement those interfaces. That's because only the properties get sent over in message passing, not the complete types